src/stream/interval.rs

```diff
@@ -71,125 +71,8 @@ impl Stream for Interval {
         if Pin::new(&mut self.delay).poll(cx).is_pending() {
             return Poll::Pending;
         }
-        let when = Instant::now();
-        let next = next_interval(when, Instant::now(), self.interval);
-        self.delay.reset(next);
+        let interval = self.interval;
+        self.delay.reset(interval);
         Poll::Ready(Some(()))
     }
 }
-
-/// Converts Duration object to raw nanoseconds if possible
-///
-/// This is useful to divide intervals.
-///
-/// While technically for large duration it's impossible to represent any
-/// duration as nanoseconds, the largest duration we can represent is about
-/// 427_000 years. Large enough for any interval we would use or calculate in
-/// async-std.
-fn duration_to_nanos(dur: Duration) -> Option<u64> {
-    dur.as_secs()
-        .checked_mul(1_000_000_000)
-        .and_then(|v| v.checked_add(u64::from(dur.subsec_nanos())))
-}
-
-fn next_interval(prev: Instant, now: Instant, interval: Duration) -> Instant {
-    let new = prev + interval;
-    if new > now {
-        return new;
-    }
-
-    let spent_ns = duration_to_nanos(now.duration_since(prev)).expect("interval should be expired");
-    let interval_ns =
-        duration_to_nanos(interval).expect("interval is less that 427 thousand years");
-    let mult = spent_ns / interval_ns + 1;
-    assert!(
-        mult < (1 << 32),
-        "can't skip more than 4 billion intervals of {:?} \
-         (trying to skip {})",
-        interval,
-        mult
-    );
-    prev + interval * (mult as u32)
-}
```

Comment:
I don't understand this logic, but is it safe to remove it?

Reply:
Yes. The logic is complex and there is not much documentation for `next_interval` function. But, as the name suggests, it is trying to find the next `Instant` to update the `Delay` with. As the new version of `Delay` already takes `Duration` as argument, we don't have to go through that complex logic to find the next `Instant`.